src/libcore/str/mod.rs

```diff
@@ -2384,11 +2384,12 @@ impl str {
         unsafe { Slices { str: self }.slice }
     }
 
-    /// Converts a mutable string slice to a mutable byte slice. To convert the
-    /// mutable byte slice back into a mutable string slice, use the
-    /// [`str::from_utf8_mut`] function.
+    /// Converts a mutable string slice to a mutable byte slice.
     ///
-    /// [`str::from_utf8_mut`]: ./str/fn.from_utf8_mut.html
+    /// # Safety
+    ///
+    /// The caller must ensure that the content of the slice remains
+    /// valid UTF-8.
```

Comment:
Per https://github.com/rust-lang/rust/issues/71033, I think this is actually (potentially) too strict. We should be able to say something like "the slice must be valid UTF-8 when its lifetime ends" or something like that.

cc @RalfJung

Reply:
Updated to reflect the change from #71033. Any suggestions for clearer or more precise wording are welcome.